ios/RNAccengageModule/RNAccengageModule.m

```diff
@@ -98,6 +103,120 @@ @implementation RNAccengageModule
+        reject(operationCode,errorMessage,nil);
+    }];
+}
+
+
+//Get Accengage inbox
+//@success BMA4SInBox
+//@failure BMA4SInBoxLoadingResult
+//
+- (void)getAccengageInboxWithSuccess:(void (^)(BMA4SInBox *inbox))success failure:(void (^)(BMA4SInBoxLoadingResult result))failure
+{
+    [BMA4SInBox obtainMessagesWithCompletionHandler:^(BMA4SInBoxLoadingResult result, BMA4SInBox *inbox) {
+        if(result != BMA4SInBoxLoadingResultLoaded)
+        {
+            failure(result);
+        }else{
+            success(inbox);
+        }
+    }];
+}
+
+//Get Message list
+//@success return a maximun of 10 messages
+//@failure BMA4SInBoxLoadingResult
+//
+- (void)getAccengageMessagesWithSuccess:(void (^)(NSMutableArray *messages))success failure:(RCTPromiseRejectBlock)failure
+{
+    _messageList = [NSMutableArray new];
+    
+    if(_numLoadedMessages < _inboxMessageList.size)
+    {
+        int unreadMessagesCount = (int)(_inboxMessageList.size - _numLoadedMessages);
+        _numLoadedMessages = MIN(unreadMessagesCount, 10);
+        
+        for (int i = 0; i < _numLoadedMessages; i++)
+        {
+            [_messageList addObject:[NSNull null]];
+            [self getInboxMessageAtIndex:i+1 messageCallback:^(NSArray *response) {
+                //override object at index
+                [_messageList setObject:response.firstObject atIndexedSubscript:i];
+                
+                //if the message is the last of the array
+                if(_messageList.count == i)
+                {
+                    success(_messageList);
+                }
+            } rejecter:^(NSString *code, NSString *message, NSError *error) {
+                failure(code,message,error);
+            }];
+        }
+    }else{
+        success([NSMutableArray new]);
+    }
+}
+
+RCT_EXPORT_METHOD(
+                  getInboxMessageAtIndex:(int)index
+                  messageCallback:(RCTResponseSenderBlock)callback
+                  rejecter:(RCTPromiseRejectBlock)reject
+){
+    //
+    //Check if the Inbox message list exists
+    //
+    if(_inboxMessageList != nil)
+    {
+        NSUInteger nsi = (NSUInteger) index;
+        
+        [_inboxMessageList obtainMessageAtIndex:nsi loaded:^(BMA4SInBoxMessage *message, NSUInteger requestedIndex) {
+        
+            //Create Message Dictionary
+            NSDictionary *messageData = @{@"title"       : message.title,
+                                          @"body"        : message.text,
+                                          @"timestamp"   : message.date,
+                                          @"category"    : message.category,
+                                          @"sender"      : message.from,
+                                          @"read"        : [NSNumber numberWithBool:message.isRead],
+                                          @"archived"    : [NSNumber numberWithBool:message.isArchived]
+                                          };
```

Comment:
customParameters are missing here

Reply:
I added to the list:
//Create Message Dictionary
            NSDictionary *messageData = @{@"title"       : message.title,
                                          @"body"        : message.text,
                                          @"timestamp"   : message.date,
                                          @"category"    : message.category,
                                          @"sender"      : message.from,
                                          @"read"        : [NSNumber numberWithBool:message.isRead],
                                          @"archived"    : [NSNumber numberWithBool:message.isArchived],
                                          @"customParameters" : message.customParams
                                          };